--- a/net/minecraft/gametest/framework/GameTestHelper.java
+++ b/net/minecraft/gametest/framework/GameTestHelper.java
@@ -77,7 +_,9 @@
 import net.minecraft.world.phys.Vec3;
 import org.jspecify.annotations.Nullable;
 
-public class GameTestHelper {
+import net.neoforged.neoforge.common.extensions.GameTestHelperExtension;
+
+public class GameTestHelper implements GameTestHelperExtension {
     public final GameTestInfo testInfo;
     private boolean finalCheckAdded;
 
