--- a/net/minecraft/client/gui/font/providers/GlyphProviderType.java
+++ b/net/minecraft/client/gui/font/providers/GlyphProviderType.java
@@ -7,8 +_,11 @@
 import net.neoforged.api.distmarker.Dist;
 import net.neoforged.api.distmarker.OnlyIn;
 
+import net.neoforged.fml.common.asm.enumextension.IExtensibleEnum;
+
 @OnlyIn(Dist.CLIENT)
-public enum GlyphProviderType implements StringRepresentable {
+@net.neoforged.fml.common.asm.enumextension.NamedEnum
+public enum GlyphProviderType implements StringRepresentable, IExtensibleEnum {
     BITMAP("bitmap", BitmapProvider.Definition.CODEC),
     TTF("ttf", TrueTypeGlyphProviderDefinition.CODEC),
     SPACE("space", SpaceProvider.Definition.CODEC),
@@ -31,5 +_,9 @@
 
     public MapCodec<? extends GlyphProviderDefinition> mapCodec() {
         return this.codec;
+    }
+
+    public static net.neoforged.fml.common.asm.enumextension.ExtensionInfo getExtensionInfo() {
+        return net.neoforged.fml.common.asm.enumextension.ExtensionInfo.nonExtended(GlyphProviderType.class);
     }
 }
