--- a/net/minecraft/world/level/block/BucketPickup.java
+++ b/net/minecraft/world/level/block/BucketPickup.java
@@ -9,8 +_,14 @@
 import net.minecraft.world.level.block.state.BlockState;
 import org.jspecify.annotations.Nullable;
 
-public interface BucketPickup {
+import net.neoforged.neoforge.common.extensions.IBucketPickupExtension;
+
+public interface BucketPickup extends IBucketPickupExtension {
     ItemStack pickupBlock(@Nullable LivingEntity user, LevelAccessor level, BlockPos pos, BlockState state);
 
+    /**
+     * Neo: use state sensitive variant instead.
+     */
+    @Deprecated
     Optional<SoundEvent> getPickupSound();
 }
