--- a/net/minecraft/data/loot/LootTableSubProvider.java
+++ b/net/minecraft/data/loot/LootTableSubProvider.java
@@ -4,7 +_,9 @@
 import net.minecraft.resources.ResourceKey;
 import net.minecraft.world.level.storage.loot.LootTable;
 
+import net.neoforged.neoforge.common.extensions.LootTableSubProviderExtension;
+
 @FunctionalInterface
-public interface LootTableSubProvider {
+public interface LootTableSubProvider extends LootTableSubProviderExtension {
     void generate(BiConsumer<ResourceKey<LootTable>, LootTable.Builder> output);
 }
